pkg/cmd/server/kubernetes/network/network.go

```diff
@@ -232,17 +236,28 @@ func (c *NetworkConfig) RunProxy() {
 }
 
 // getNodeIP is copied from the upstream proxy config to retrieve the IP of a node.
-func getNodeIP(client kv1core.CoreV1Interface, hostname string) net.IP {
-	var nodeIP net.IP
-	node, err := client.Nodes().Get(hostname, metav1.GetOptions{})
-	if err != nil {
-		glog.Warningf("Failed to retrieve node info: %v", err)
-		return nil
+func getNodeIP(client kv1core.CoreV1Interface, hostname string) (net.IP, error) {
+	var node *v1.Node
+	var nodeErr error
+
+	// This will wait 2, 4, 8, ... 256 seconds, for a total of 8.5 mins
+	nodeWaitBackoff := utilwait.Backoff{
+		Duration: 2 * time.Second,
+		Factor:   2,
+		Steps:    8,
 	}
-	nodeIP, err = utilnode.GetNodeHostIP(node)
+	utilwait.ExponentialBackoff(nodeWaitBackoff, func() (bool, error) {
+		node, nodeErr = client.Nodes().Get(hostname, metav1.GetOptions{})
+		return nodeErr == nil, nil
```

Comment:
Ideally, 'return nodeErr==nil, nodeErr'

Reply:
I can't... per vendor/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go if I return an error, the backoff ends and the error is returned.

The first `nodeErr == nil` is the backoff termination condition, and we want that to be true when there are no errors.  I update the nodeErr defined outside the condition function so that we can get the error without messing with the backoff termination.

Reply:
Done, thanks